### ISSUE TITLE
Plugin error from gemelli:    "['sample-id'] not found in axis"

### ISSUE THREAD

**Issue by govind-sah** [biocore/gemelli#50]
Hi,
Please ignore if is not easy to follow as this is my first issue on this forum.

**Files used for this analysis**
[Archive.zip](https://github.com/biocore/gemelli/files/7239252/Archive.zip)

I am performing 16s rRNA analysis through qiime2. I want to study the longitudinal distribution of features over the period of sample collection (sampling-day). In my metadata file, I have 'sample-id' as first column however

**when I run the following command for gemelli analysis**
qiime gemelli ctf --i-table lab_wise_analysis/bacteriology/bacteriology_table.qza --m-sample-metadata-file lab_wise_analysis/bacteriology/metadata_bacteriology.tsv --m-feature-metadata-file taxonomy.qza --p-state-column sampling-day --p-individual-id-column sample-id --output-dir lab_wise_analysis/bacteriology/CTF-results-GEMELLI

**it gives me following error:** 
Plugin error from gemelli:
  "['sample-id'] not found in axis"
Debug info has been saved to /var/folders/31/7jhpcwd91kq6rmrbrjdqtvm40000gn/T/qiime2-q2cli-err-3f1eij0c.log

**When I view the debug info, it says**
_Traceback (most recent call last):
  File "/Users/pridelab/opt/miniconda3/envs/qiime2-2021.4/lib/python3.8/site-packages/q2cli/commands.py", line 329, in __call__
    results = action(**arguments)
  File "<decorator-gen-540>", line 2, in ctf
  File "/Users/pridelab/opt/miniconda3/envs/qiime2-2021.4/lib/python3.8/site-packages/qiime2/sdk/action.py", line 244, in bound_callable
    outputs = self._callable_executor_(scope, callable_args,
  File "/Users/pridelab/opt/miniconda3/envs/qiime2-2021.4/lib/python3.8/site-packages/qiime2/sdk/action.py", line 390, in _callable_executor_
    output_views = self._callable(**view_args)
  File "/Users/pridelab/opt/miniconda3/envs/qiime2-2021.4/lib/python3.8/site-packages/gemelli/ctf.py", line 30, in ctf
    state_ordn, ord_res, dists, straj, ftraj = ctf_helper(table,
  File "/Users/pridelab/opt/miniconda3/envs/qiime2-2021.4/lib/python3.8/site-packages/gemelli/ctf.py", line 71, in ctf_helper
    all_sample_metadata = sample_metadata.drop(keep_cols, axis=1)
  File "/Users/pridelab/opt/miniconda3/envs/qiime2-2021.4/lib/python3.8/site-packages/pandas/core/frame.py", line 4308, in drop
    return super().drop(
  File "/Users/pridelab/opt/miniconda3/envs/qiime2-2021.4/lib/python3.8/site-packages/pandas/core/generic.py", line 4153, in drop
    obj = obj._drop_axis(labels, axis, level=level, errors=errors)
  File "/Users/pridelab/opt/miniconda3/envs/qiime2-2021.4/lib/python3.8/site-packages/pandas/core/generic.py", line 4188, in _drop_axis
    new_axis = axis.drop(labels, errors=errors)
  File "/Users/pridelab/opt/miniconda3/envs/qiime2-2021.4/lib/python3.8/site-packages/pandas/core/indexes/base.py", line 5591, in drop
    raise KeyError(f"{labels[mask]} not found in axis")

[](url)
KeyError: "['sample-id'] not found in axis"_

**Could someone please help as I am a beginner in this field.**
Thank you

**Comment by govind-sah** [comment]
Thank you so much @ElDeveloper for responding so quickly to my issue. I used `lab-location` as the  `--p-individual-id-column` and it fixed the issue. I did not added an additional column such as `individual-id` and it still worked.
Regarding, GEMELLI suitability for my dataset, I have multiple samples collected from each sampling site: `bench`, `floor` and `sink` over a period of time (day/dates). That's the reason I wanted to study the microbial diversity over the period of sampling dates/day using a longitudinal analysis tool such as `Gemelli `. 
As I have mentioned above, I am new to this field and still trying to learn things so might be completely wrong on this.
Thank you,
Govind
